explorer/src/components/account/AnchorAccountCard.tsx

```diff
@@ -1,63 +1,62 @@
-import BN from "bn.js";
-
-import ReactJson from "react-json-view";
-import { useCluster } from "providers/cluster";
 import { useAnchorProgram } from "providers/anchor";
 
 export function AnchorAccountCard({ account }: { account: Account }) {
+  const { lamports } = account;
   const { url } = useCluster();
-  const program = useAnchorProgram(
-    account.details?.owner.toString() ?? "",
+  const anchorProgram = useAnchorProgram(
+    account.details?.owner.toString() || "",
     url
   );
+  const rawData = account?.details?.rawData;
+  const programName = getProgramName(anchorProgram) || "Unknown Program";
 
-  const { foundAccountLayoutName, decodedAnchorAccountData } = useMemo(() => {
-    let foundAccountLayoutName: string | undefined;
-    let decodedAnchorAccountData: { [key: string]: any } | undefined;
-    if (program && account.details && account.details.rawData) {
-      const accountBuffer = account.details.rawData;
-      const discriminator = accountBuffer.slice(0, 8);
-
-      // Iterate all the structs, see if any of the name-hashes match
-      Object.keys(program.account).forEach((accountType) => {
-        const layoutName = capitalizeFirstLetter(accountType);
-        const discriminatorToCheck =
-          BorshAccountsCoder.accountDiscriminator(layoutName);
-
-        if (discriminatorToCheck.equals(discriminator)) {
-          foundAccountLayoutName = layoutName;
-          const accountDecoder = program.account[accountType];
-          decodedAnchorAccountData = accountDecoder.coder.accounts.decode(
-            layoutName,
-            accountBuffer
-          );
-        }
-      });
+  const { decodedAccountData, accountDef } = useMemo(() => {
+    let decodedAccountData: any | null = null;
+    let accountDef: IdlTypeDef | undefined = undefined;
+    if (anchorProgram && rawData) {
+      const coder = new BorshAccountsCoder(anchorProgram.idl);
+      const accountDefTmp = anchorProgram.idl.accounts?.find(
+        (accountType: any) =>
+          (rawData as Buffer)
+            .slice(0, 8)
+            .equals(BorshAccountsCoder.accountDiscriminator(accountType.name))
+      );
+      if (accountDefTmp) {
+        accountDef = accountDefTmp;
+        decodedAccountData = coder.decode(accountDef.name, rawData);
+      }
     }
-    return { foundAccountLayoutName, decodedAnchorAccountData };
-  }, [program, account.details]);
 
-  if (!foundAccountLayoutName || !decodedAnchorAccountData) {
+    return {
+      decodedAccountData,
+      accountDef,
+    };
+  }, [anchorProgram, rawData]);
+
+  if (lamports === undefined) return null;
+  if (!anchorProgram) return <ErrorCard text="No Anchor IDL found" />;
+  if (!decodedAccountData || !accountDef) {
     return (
-      <ErrorCard text="Failed to decode account data according to its public anchor interface" />
+      <ErrorCard text="Failed to decode account data according to the public Anchor interface" />
     );
   }
 
   return (
-    <>
+    <div>
```

Comment:
Why this extra level of DOM nesting rather than returning these nodes in a fragment like before?